defaults/validate_test.go

```diff
@@ -371,6 +373,50 @@ func TestValidate(t *testing.T) {
 		err := Validate(cfg{"start"})
 		require.Error(t, err)
 	})
+
+	t.Run("empty emptyorfile", func(t *testing.T) {
+		var cfg serverConfiguration
+		Set(&cfg)
+
+		cfg.LicenseFile = ""
+
+		err := Validate(&cfg)
+		require.NoError(t, err)
+	})
+
+	t.Run("valid non-empty emptyorfile", func(t *testing.T) {
+		var cfg serverConfiguration
+		Set(&cfg)
+
+		// We need a file that exists, so let's use the path to the executable running
+		// this test, which is guaranteed to exist when the test is running
+		f, err := os.Executable()
+		require.NoError(t, err)
+		cfg.LicenseFile = f
+
+		err = Validate(&cfg)
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid emptyorfile", func(t *testing.T) {
+		var cfg serverConfiguration
+		Set(&cfg)
+
+		cfg.LicenseFile = "/invalid/path/to/inexistent/file"
+
+		err := Validate(&cfg)
+		require.Error(t, err)
+	})
+
+	t.Run("not a path for emptyorfile", func(t *testing.T) {
+		var cfg serverConfiguration
+		Set(&cfg)
+
+		cfg.LicenseFile = "not a file"
+
+		err := Validate(&cfg)
+		require.Error(t, err)
```

Comment:
nit: I usually find value in using [`EqualError`](https://pkg.go.dev/github.com/stretchr/testify/require#EqualError) rather than simply checking if it errored. It saved me from a few bugs in the past :)